lib/manager/helmfile/extract.ts

```diff
@@ -0,0 +1,77 @@
+import is from '@sindresorhus/is';
+import yaml from 'js-yaml';
+
+import { logger } from '../../logger';
+import { PackageFile, PackageDependency, ExtractConfig } from '../common';
+
+export function extractPackageFile(
+  content: string,
+  fileName: string,
+  config: ExtractConfig
+): PackageFile {
+  let deps = [];
+  let doc;
+  const aliases: Record<string, string> = {};
+  try {
+    doc = yaml.safeLoad(content, { json: true });
+  } catch (err) {
+    logger.debug({ fileName }, 'Failed to parse helmfile helmfile.yaml');
+    return null;
+  }
+  if (!(doc && is.array(doc.releases))) {
+    logger.debug({ fileName }, 'helmfile.yaml has no releases');
+    return null;
+  }
+
+  if (doc.repositories) {
+    for (let i = 0; i < doc.repositories.length; i += 1) {
+      aliases[doc.repositories[i].name] = doc.repositories[i].url;
+    }
+  }
+  logger.debug({ aliases }, 'repositories discovered.');
+
+  deps = doc.releases.map(dep => {
+    let depName = dep.chart;
+    let repoName = null;
+
+    if (dep.chart.startsWith('./')) {
+      return {
+        depName,
+        skipReason: 'local-chart',
+      } as PackageDependency;
+    }
+
+    if (dep.chart.includes('/')) {
+      const v = dep.chart.split('/');
+      depName = v[1];
+      repoName = v[0];
+    } else {
+      repoName = dep.chart;
+    }
+
+    const res: PackageDependency = {
+      depName,
+      currentValue: dep.version,
+      registryUrls: [aliases[repoName]]
+        .concat([config.aliases[repoName]])
+        .filter((v, _) => {
+          return v != null;
+        }),
+    };
+
+    if (res.depName.includes('{') || res.depName.includes('}')) {
+      res.skipReason = 'invalid-chart';
+    }
+
+    if (is.emptyArray(res.registryUrls)) {
+      res.skipReason = 'invalid-registry';
```

Comment:
There's no "default" registry? i.e. it needs to be included in every dependency definition?

Reply:
I've just done a test and is actually taking back the one configured in `helm`, I get misled by seeing always the stable definition in the `helmfile.yaml`

In the case of `renovate` the default should come from `config.aliases`

Reply:
If it works for you and generates PRs with "typical" helmfiles, then it's ok by me.

Reply:
I guess I can change the error as `unknown-registry` in order to make more visible that the problem is not the syntax but the fact that we lack the information needed to select the proper registry to be used.